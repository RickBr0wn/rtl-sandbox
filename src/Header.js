import React from 'react'
import PropTypes from 'prop-types'

function Header({ text }) {
  return (
    <div>
      <h2 test-id="title" className="title">
        {text}
      </h2>
    </div>
  )
}

Header.propTypes = {
  text: PropTypes.string.isRequired,
}

export default Header
