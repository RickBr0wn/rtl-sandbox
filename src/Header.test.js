import React from 'react'
import {
  render,
  cleanup,
  fireEvent,
  waitForElement
} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
// the axios mock is in __mocks__/
// see https://jestjs.io/docs/en/manual-mocks
// import axiosMock from 'axios'
import Header from './Header'

afterEach(cleanup)

it('should render', () => {
  const { asFragment } = render(<Header text='Hello!' />)
  expect(asFragment()).toMatchSnapshot()
})

it('should insert text in h1', () => {
  const { getByTestId, getByText } = render(<Header text='Hello!' />)
  expect(getByTestId('title')).toHaveTextContent('Hello!')
  expect(getByText('Hello!')).toHaveClass('title')
})
