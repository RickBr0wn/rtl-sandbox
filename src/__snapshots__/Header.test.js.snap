// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="title"
      test-id="title"
    >
      Hello!
    </h2>
  </div>
</DocumentFragment>
`;
